### PR TITLE
Record reStructuring

### PR DESCRIPTION
**Pull request by MartinSchmidt**
All records now use the same format, required attributes.
Removed all model access to RegistryId, this only exists on repository level for now, slated for later removal.
Simplified code using Registry not having to lookup the name each time.

### CODE REVIEW

File: src/ProjectOrigin.WalletSystem.IntegrationTests/QueryCertificatesTest.cs [hunk 1]
```diff
@@ -56,37 +56,111 @@ public async void QueryCertificates()
...
-                await registryRepository.InsertRegistry(registry);
 
                 var attributes = new List<CertificateAttribute>
                 {
-                    new ("AssetId", "571234567890123456"),
-                    new ("TechCode", "T070000"),
-                    new ("FuelCode", "F00000000")
+                    new (){Key="AssetId", Value= "571234567890123456"},
+                    new (){Key="TechCode", Value= "T070000"},
+                    new (){Key="FuelCode", Value= "F00000000"}
```
Comment (duizer) [inline src/ProjectOrigin.WalletSystem.IntegrationTests/QueryCertificatesTest.cs:84]: Formatting of these lines is inconsistent.

File: src/ProjectOrigin.WalletSystem.Server/Repositories/CertificateRepository.cs [hunk 2]
```diff
@@ -13,34 +13,42 @@ public class CertificateRepository : ICertificateRepository
...
-    {
-        this._connection = connection;
-    }
+    public CertificateRepository(IDbConnection connection) => this._connection = connection;
 
-    public Task InsertSlice(Slice newSlice)
+    public async Task InsertSlice(Slice newSlice)
     {
-        return _connection.ExecuteAsync(@"INSERT INTO Slices(Id, DepositEndpointId, DepositEndpointPosition, RegistryId, CertificateId, Quantity, RandomR, SliceState) VALUES (@id, @depositEndpointId, @depositEndpointPosition, @registryId, @certificateId, @quantity, @randomR, @sliceState)", new { newSlice.Id, newSlice.DepositEndpointId, newSlice.DepositEndpointPosition, newSlice.RegistryId, newSlice.CertificateId, newSlice.Quantity, newSlice.RandomR, newSlice.SliceState });
+        var registryId = await _connection.QuerySingleAsync<Guid>("SELECT Id FROM Registries WHERE Name = @registry", new { newSlice.Registry });
```
Comment (duizer) [inline src/ProjectOrigin.WalletSystem.Server/Repositories/CertificateRepository.cs:20]: I thought the goal was to drop the Registries table

File: src/ProjectOrigin.WalletSystem.IntegrationTests/WalletSystemTestsBase.cs [hunk 3]
```diff
@@ -64,25 +64,39 @@ protected async Task<DepositEndpoint> CreateWalletDepositEndpoint(string owner)
...
         {
             var certificateRepository = new CertificateRepository(connection);
             var attributes = new List<CertificateAttribute>
             {
-                new ("AssetId", "571234567890123456"),
-                new ("TechCode", "T070000"),
-                new ("FuelCode", "F00000000")
+                new () {Key="AssetId", Value= "571234567890123456"},
+                new () {Key="TechCode", Value= "T070000"},
+                new () {Key="FuelCode", Value= "F00000000"}
```
Comment (duizer) [inline src/ProjectOrigin.WalletSystem.IntegrationTests/WalletSystemTestsBase.cs:88]: Formatting

File: src/ProjectOrigin.WalletSystem.IntegrationTests/WalletSystemTestsBase.cs [hunk 4]
```diff
@@ -94,39 +108,35 @@ protected async Task InsertSlice(DepositEndpoint depositEndpoint, int position,
...
-
-        await certificateRepository.InsertSlice(receivedSlice);
-    }
-
-    protected static async Task<RegistryModel> GetOrInsertRegistry(NpgsqlConnection connection, string registryName)
-    {
-        var registryRepository = new RegistryRepository(connection);
-        var registry = await registryRepository.GetRegistryFromName(registryName);
-        if (registry is null)
+        var receivedSlice = new Slice
```
Comment (duizer) [inline src/ProjectOrigin.WalletSystem.IntegrationTests/WalletSystemTestsBase.cs:128]: Could `receivedSlice` be renamed to `slice`?

File: src/ProjectOrigin.WalletSystem.IntegrationTests/Repositories/CertificateRepositoryTests.cs [hunk 5]
```diff
@@ -26,38 +26,41 @@ public CertificateRepositoryTests(PostgresDatabaseFixture dbFixture) : base(dbFi
...
-        var registry = await CreateRegistry();
+        var registry = _fixture.Create<string>();
         var attributes = new List<CertificateAttribute>
         {
-            new ("AssetId", "571234567890123456"),
-            new ("TechCode", "T070000"),
-            new ("FuelCode", "F00000000")
+            new (){ Key="AssetId", Value= "571234567890123456"},
+            new (){ Key="TechCode", Value= "T070000"},
+            new (){ Key="FuelCode", Value= "F00000000"}
```
Comment (duizer) [inline src/ProjectOrigin.WalletSystem.IntegrationTests/Repositories/CertificateRepositoryTests.cs:34]: Formatting

File: src/ProjectOrigin.WalletSystem.Server/Repositories/CertificateRepository.cs (hunk 2 above)
Comment (MartinSchmidt) [reply src/ProjectOrigin.WalletSystem.Server/Repositories/CertificateRepository.cs:20]: Well i concentrated in this PR simply to remove all references to it outside of the repository.
Then we can remove the table and denormalise all the data in another PR later, this already made quality of life a lot better.

File: src/ProjectOrigin.WalletSystem.IntegrationTests/QueryCertificatesTest.cs (hunk 1 above)
Comment (MartinSchmidt) [reply src/ProjectOrigin.WalletSystem.IntegrationTests/QueryCertificatesTest.cs]: fixed

File: src/ProjectOrigin.WalletSystem.IntegrationTests/Repositories/CertificateRepositoryTests.cs (hunk 5 above)
Comment (MartinSchmidt) [reply src/ProjectOrigin.WalletSystem.IntegrationTests/Repositories/CertificateRepositoryTests.cs]: fixed

File: src/ProjectOrigin.WalletSystem.Server/Repositories/CertificateRepository.cs (hunk 2 above)
Comment (duizer) [reply src/ProjectOrigin.WalletSystem.Server/Repositories/CertificateRepository.cs:20]: Cool

**Review by duizer — APPROVED** [review]

**Pull request merged**